src/ray/common/status.h

```diff
@@ -47,34 +48,35 @@ class error_code;
 }  // namespace boost
 
 // Return the given status if it is not OK.
-#define RAY_RETURN_NOT_OK(s)           \
-  do {                                 \
-    const ::ray::Status &_s = (s);     \
-    if (RAY_PREDICT_FALSE(!_s.ok())) { \
-      return _s;                       \
-    }                                  \
+#define RAY_RETURN_NOT_OK(s)                                \
+  do {                                                      \
+    const ::ray::Status &RAY_UNIQUE_VARIABLE(_s) = (s);     \
```

Comment:
why do we need this instead of _s ?

Reply:
People could name variable as `_s`; for temporary variables we use internally, we should guarantee (at best) its uniqueness to avoid shadowing.

Reply:
even if people has a variable `_s` it's shadowed by this new declaration in the do-while block?

Reply:
No, `RAY_UNIQUE_VARIABLE` macros creates a "unique" variable with line number in var name.

Reply:
I mean, why do we want to avoid this shadowing? Since there will be no risk of accidental use of a user _s variable.

On the other hand, even if we want to avoid shadowing, the risk of shadowing a `_s123` is equivalent to the risk of shadowing a `_s`: not possible because of our code reviews.

Reply:
> I mean, why do we want to avoid this shadowing? Since there will be no risk of accidental use of a user _s variable.

Variable shadowing is discouraged, I met quite a few issues in the past.
I sometimes checked processed code after preprocessing stage to check macro and template effect.

> On the other hand, even if we want to avoid shadowing, the risk of shadowing a _s123 is equivalent to the risk of shadowing a _s: not possible because of our code reviews.

That's why I say "at best".

Anyway I don't think it's worth arguing over, change reverted. PTAL